2.4/examples/replica/mongodb-clustered-petset.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: mongodb-petset-replica
+  annotations:
+    description: "MongoDB Replication Example (based on PetSet)"
+    iconClass: "icon-mongodb"
+    tags: "database,mongodb,replication"
+parameters:
+  - name: MONGODB_USER
+    description: "The username for MongoDB account to be created"
+    generate: expression
+    from: "[a-zA-Z0-9]{3}"
+  - name: MONGODB_PASSWORD
+    description: "Password for the MongoDB user account"
+    generate: expression
+    from: "[a-zA-Z0-9]{12}"
+  - name: MONGODB_DATABASE
+    description: "Name for the MongoDB database"
+    value: sampledb
+  - name: MONGODB_ADMIN_PASSWORD
+    description: "Password for the MongoDB admin user"
+    generate: expression
+    from: "[a-zA-Z0-9]{12}"
+  - name: MONGODB_REPLICA_NAME
+    description: "The name of the replica set"
+    value: rs0
+  # It must match with Service.metadata.name and PetSet.spec.serviceName
+  - name: MONGODB_SERVICE_NAME
+    description: "The name of the MongoDB Service (used to DNS lookup)"
+    value: mongodb
+  - name: MONGODB_KEYFILE_VALUE
+    description: "The value of the MongoDB Key (see for details: https://docs.mongodb.org/v2.4/tutorial/generate-key-file/)"
+    generate: expression
+    from: "[a-zA-Z0-9]{255}"
```

Comment:
we should start using the displayname field in our parameters too: https://github.com/openshift/origin/blob/master/examples/quickstarts/dancer-mysql.json#L449-L453